openerp/addons/base/ir/ir_model.py

```diff
@@ -324,21 +327,23 @@ def _drop_column(self, cr, uid, ids, context=None):
         for field in self.browse(cr, uid, ids, context):
             if field.name in MAGIC_COLUMNS:
                 continue
-            model = self.pool[field.model]
-            cr.execute('select relkind from pg_class where relname=%s', (model._table,))
+            table_name = field.model_id.table_name
+            cr.execute('select relkind from pg_class where relname=%s', (table_name,))
             result = cr.fetchone()
-            cr.execute("SELECT column_name FROM information_schema.columns WHERE table_name ='%s' and column_name='%s'" %(model._table, field.name))
+            cr.execute("SELECT column_name FROM information_schema.columns WHERE table_name=%s and column_name=%s", (table_name, field.name))
             column_name = cr.fetchone()
             if column_name and (result and result[0] == 'r'):
-                cr.execute('ALTER table "%s" DROP column "%s" cascade' % (model._table, field.name))
-            # remove m2m relation table for custom fields
-            # we consider the m2m relation is only one way as it's not possible
-            # to specify the relation table in the interface for custom fields
-            # TODO master: maybe use ir.model.relations for custom fields
-            if field.state == 'manual' and field.ttype == 'many2many':
-                rel_name = model._fields[field.name].relation
-                cr.execute('DROP table "%s"' % (rel_name))
-            model._pop_field(field.name)
+                cr.execute('ALTER table "%s" DROP column "%s" cascade' % (table_name, field.name))
+            model = self.pool.get(field.model)
+            if model:
+            	# remove m2m relation table for custom fields
+            	# we consider the m2m relation is only one way as it's not possible
+            	# to specify the relation table in the interface for custom fields
+            	# TODO master: maybe use ir.model.relations for custom fields
+            	if field.state == 'manual' and field.ttype == 'many2many':
+                    rel_name = model._fields[field.name].relation
+                    cr.execute('DROP table "%s"' % (rel_name))
+                model._pop_field(field.name)
```

Comment:
Please fix the indentation of this block: 3 or 4 spaces on the right of the "if" at line 338.